esmvalcore/cmor/_fixes/fix.py

```diff
@@ -5,29 +5,43 @@
 import inspect
 import tempfile
 from pathlib import Path
+from typing import TYPE_CHECKING, Optional
 
 from ..table import CMOR_TABLES
 
+if TYPE_CHECKING:
+    from ...config import Session
+    from ..table import VariableInfo
```

Comment:
I'd really make this imports use the `esmvalcore` package than do the up, up and up dir

Reply:
it shouldn't since circ imports issues happen when one module imports another module that imports the first module, but OK - I'd have changed all those .. and . imports with package-level imports but we'd prob have to go elsewhere do that too, outside the scope of this PR :+1:

Reply:
Indeed we may want to consider using more absolute imports, though I like relative imports if they're importing from a module at the same level and are deeply nested. From [PEP8](https://peps.python.org/pep-0008/#imports):

Absolute imports are recommended, as they are usually more readable and tend to be better behaved (or at least give better error messages) if the import system is incorrectly configured (such as when a directory inside a package ends up on sys.path):
```python
import mypkg.sibling
from mypkg import sibling
from mypkg.sibling import example
```
However, explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose:
```python
from . import sibling
from .sibling import example
```